src/net/conn_ssl.c

```diff
@@ -0,0 +1,148 @@
+#include <postgres.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include <unistd.h>
+#include <pg_config.h>
+
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+
+#include "conn_internal.h"
+#include "conn_plain.h"
+
+typedef struct SSLConnection
+{
+	Connection	conn;
+	SSL_CTX    *ssl_ctx;
+	SSL		   *ssl;
+} SSLConnection;
+
+static int
+ssl_setup(SSLConnection *conn)
+{
+	int			ret;
+
+	conn->ssl_ctx = SSL_CTX_new(SSLv23_method());
+
+	if (NULL == conn->ssl_ctx)
+		elog(ERROR, "connection library: could not create SSL context");
```

Comment:
In general- I think a more descriptive prefix would be a good choice for these errors, ie TimescaleDB telemetry connection lib: blah blah blah 
Don't want people to be confused by connection library things that are different from normal postgres connections.

Reply:
A systematic prefix is probably a good idea, but I'd make it short albeit identifiable, such as `ts_net` or `ts::net`. Then we could have, e.g., `ts::net`, `ts::telemetry`, `ts:bgw` etc.